crowbar_framework/lib/crowbar/deployment_queue.rb

```diff
@@ -130,29 +129,31 @@ def process_queue
         begin
           f = file_lock.acquire("queue", logger: logger)
 
-          if proposal_queue.empty?
+          if @proposal_queue.nil?
             logger.debug("process queue: exit: empty queue")
             return
           end
 
-          logger.debug("process queue: queue: #{proposal_queue.proposals.inspect}")
+          logger.debug("process queue: queue: #{@proposal_queue.inspect}")
 
           # Test for ready
           remove_list = []
-          proposal_queue.proposals.each do |item|
-            prop = Proposal.where(barclamp: item["barclamp"], name: item["inst"]).first
+          ProposalQueue.all.each do |item|
```

Comment:
But that is never guaranteed, so please add it directly here or create a named scope ```ordered``` that you can call for get the FIFO record for sure, e.g.

```ruby
  scope :ordered, lambda {
    order("proposal_queue.created_at ASC")
  }
```